CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# Contributing to HClib
+
+Before opening a pull request,
+please be sure to read through these guidelines.
+Following the guidelines listed below will make it much more likely
+for your contributions to be quickly accepted and merged.
+
+## Licensing
+
+All contributions submitted to this project are governed
+by the terms outlined in our license agreement.
+Specifically, our license (Apache 2.0)
+has an explicit contributions clause in [section
+5](https://github.com/habanero-rice/hclib/blob/6ad778a/LICENSE#L131-L137):
+
+> Unless You explicitly state otherwise, any Contribution intentionally
+> submitted for inclusion in the Work by You to the Licensor shall be under
+> the terms and conditions of this License, without any additional terms or
+> conditions.  Notwithstanding the above, nothing herein shall supersede or
+> modify the terms of any separate license agreement you may have executed
+> with Licensor regarding such Contributions.
+
+## Commits
+
+Please follow established best practices for your commits.
+Do only one thing per commit, and write a descriptive commit message.
+[Chris Beams' *seven rules of a great Git commit
+message*](https://chris.beams.io/posts/git-commit/#seven-rules)
+provides a good set of guidelines:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain *what* and *why* vs. *how*
+
+## Pull Requests
+
+Please follow established best practices for your pull requests.
+Give a short description of *what* your patch does in the title,
+and include a short description of *why* the patch is needed in the comment.
+A complex title or summary is a good indication that your pull request
+is doing too many things, and should be split into multiple patches.
+
+See also [*10 tips for better Pull Requests* by Mark
+Seemann](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/).
+
+When you create a pull request, please assign one or more core team members as 
```

Comment:
Okay, I agree with adding the first part to the text. But the second part seems like it should just be the responsibility of the "core team" reviewing the code to judge the disruptiveness of the change and decide if any special notification is required? We want to have a rigorous process but also not an onerous one, otherwise we de-incentivize contributions. Asking someone to figure out everyone who's committed to a file even in just the last month sounds painful to me, and is likely to turn others away.